src/adagrams.js

```diff
@@ -1,7 +1,172 @@
+//  Wave 1 Problem
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const alphabets = {
+      a: 9,
+      b: 2,
+      c: 2,
+      d: 4,
+      e: 12,
+      f: 2,
+      g: 3,
+      h: 2,
+      i: 9,
+      j: 1,
+      k: 1,
+      l: 4,
+      m: 2,
+      n: 6,
+      o: 8,
+      p: 2,
+      q: 1,
+      r: 6,
+      s: 4,
+      t: 6,
+      u: 4,
+      v: 2,
+      x: 1,
+      y: 2,
+      z: 1
+    };
+    let allLetters = [];
+    for (let letter in alphabets) {
+      let quantity = alphabets[letter];
+      let letters = [];
+      let i = 0;
+      while (i < quantity) {
+        letters.push(letter);
+        i++;
+      }
+      allLetters.push(letters);
+    }
+    allLetters = allLetters.flat();
+
+    allLetters = shuffle(allLetters);
+
+    let yourLetters = [];
+    let j = 0;
+    while (j < 10) {
+      yourLetters.push(allLetters[j]);
+      j++;
+    }
+
+    return yourLetters;
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    let word = input.toLowerCase().split("");
+    let sameLetters = shuffle(lettersInHand);
+
+    word.forEach(function(letter) {
+      if (sameLetters.includes(letter)) {
+        let letterLocation = sameLetters.indexOf(letter);
+        sameLetters.splice(letterLocation, 1);
+      } else {
+        return false;
+      }
```

Comment:
I am surprised to realize 2 of the 3 tests still fail when I add the return statement mentioned below. There are several bugs in this code that it'd be good for us to talk about during our 1:1 next week.